internal/elasticsearch/cluster/script.go

```diff
@@ -0,0 +1,151 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceScript() *schema.Resource {
+	scriptSchema := map[string]*schema.Schema{
+		"script_id": {
+			Description: "Identifier for the stored script. Must be unique within the cluster.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+		},
+		"lang": {
+			Description:  "Script language. For search templates, use `mustache`. Defaults to `painless`.",
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "painless",
```

Comment:
This default makes sense in the context of a stored script, but not when used to manage a search template. IMO it would be better to mirror the API and have this as a required field without a default.

Reply:
Oh that's true, fixed to make it required field.